lib/FusionInventory/Agent/Task/Inventory/Linux/Storages.pm

```diff
@@ -83,15 +83,27 @@ sub _getDevices {
             $device->{SERIALNUMBER}
         );
 
-        if (!$device->{MANUFACTURER} or $device->{MANUFACTURER} eq 'ATA') {
+        if (!$device->{MANUFACTURER} || $device->{MANUFACTURER} eq 'ATA') {
             $device->{MANUFACTURER} = getCanonicalManufacturer(
                 $device->{MODEL}
             );
+        } elsif ($device->{MANUFACTURER} && $device->{MANUFACTURER} =~ /^0x(\w+)$/) {
+            my $vendor = getPCIDeviceVendor(id => lc($1));
+            $device->{MANUFACTURER} = $vendor->{name}
+                if $vendor && $vendor->{name};
         }
 
         if (!$device->{DISKSIZE} && $device->{TYPE} !~ /^cd/) {
             $device->{DISKSIZE} = getDeviceCapacity(device => '/dev/' . $device->{NAME});
         }
+
+        # In some case, serial can't be defined using hdparm (command missing or virtual disk)
+        # Then we can define a serial searching for few specific identifiers
+        if (!$device->{SERIALNUMBER}) {
+            $params{device} = '/dev/' . $device->{NAME};
+            $device->{SERIALNUMBER} = _getDiskIdentifier(%params) ||
+                _getPVUUID(%params) || "";
```

Comment:
I'm not sure of the usefulness of using an empty string instead of letting the variable undefined.

Reply:
Fixed in 99502b3